cpp/src/binaryop/compiled/util.cpp

```diff
@@ -181,6 +183,24 @@ std::optional<data_type> get_common_type(data_type out, data_type lhs, data_type
 
 bool is_supported_operation(data_type out, data_type lhs, data_type rhs, binary_operator op)
 {
-  return double_type_dispatcher(lhs, rhs, is_supported_operation_functor{}, out, op);
+  return double_type_dispatcher(lhs, rhs, is_supported_operation_functor{}, out, op) ||
+         (is_struct(lhs) && is_struct(rhs) && op != binary_operator::GENERIC_BINARY &&
+          op != binary_operator::INVALID_BINARY);
```

Comment:
This falls back to the explicit instantiation of the other functions. Jakes previous suggestion regarding `is_invocable` is not possible with struct type because we don't have access to the children as part of the `data_type`. I'm totally stumped on how you would do proper type checking here with just the `data_type`'s which is why there is an overload with `column_view` arguments that can be invoked later.